frontend/src/components/ImplementationReportView/EmbedBadges.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useState } from "react";
+import { Implementation } from "../../data/parseReportData";
+import CopyToClipboard from "../CopyToClipboard";
+import Dialect from "../../data/Dialect";
+import { complianceBadgeFor, versionsBadgeFor } from "../../data/Badge";
+import { badgeFormatOptions, BadgeFormatOption } from "./BadgeFormats";
+
+const EmbedBadges: React.FC<{
+  implementation: Implementation;
+}> = ({ implementation }) => {
+  const [activeTab, setActiveTab] = useState("URL");
+  const [activeBadge, setActiveBadge] = useState("JSON Schema Versions");
+  const [badgeURI, setBadgeURI] = useState(
+    versionsBadgeFor(implementation).href(),
+  );
+
+  const handleSelectBadge = (badgeName: string, URI: string): void => {
+    setActiveBadge(badgeName);
+    setBadgeURI(URI);
+  };
+
+  const handleSelectTab = (tabKey: string | null) => {
+    if (tabKey) {
+      setActiveTab(tabKey);
+    }
+  };
+
+  const results = Object.entries(implementation.results).sort((a, b) => {
+    return (
+      Dialect.withName(b[0]).firstPublicationDate.getTime() -
```

Comment:
Is the `.getTime` here actually needed? Javascript doesn't know how to just sort Dates themselves?

Reply:
My linter was showing error `The left-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type.` and `The right-hand side of an arithmetic operation must be of type 'any', 'number', 'bigint' or an enum type.` without using `.getTime()` method.

Reply:
Got it, ok, fair enough then.